### PR TITLE
Update repo locations

### PR DESCRIPTION
**Pull request by amzn-changml**
Signed-off-by: Mike Chang <changml@amazon.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ The project supports the following platforms:
 The following commands should prepare O3DE:
 
 ```
-~$ git clone https://github.com/aws-lumberyard-dev/o3de.git
+~$ git clone https://github.com/o3de/o3de.git
```
[the remark this answers is not in the document]
Comment (amzn-changml) [reply README.md:63]: Should we create a new fork of O3DE specifically for the ROSCon demo? The risk here is that the `aws-lumberyard-dev` fork may not exist indefinitely

[the remark this answers is not in the document]
Comment (amzn-pratikpa) [reply README.md:63]: @spham-amzn can confirm but I think we should be able to build and run with the O3DE development branch

[the remark this answers is not in the document]
Comment (adamdbrw) [reply README.md:63]: If this is the case, we also need to update another point where we are telling users to checkout roscon_2022 branch. We have not tested at all with O3DE development branch on our side (I guess some commits to it during ROSCon could break something)

[the remark this answers is not in the document]
Comment (amzn-pratikpa) [reply README.md:63]: Tested with O3DE development branch and it worked just fine

**Review by amzn-pratikpa — APPROVED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (amzn-changml) [reply README.md:63]: Awesome, I've updated the instructions to use the `development` branch

**Pull request merged**